azurerm/resource_arm_data_factory_v2.go

```diff
@@ -0,0 +1,343 @@
+package azurerm
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/datafactory/mgmt/2018-06-01/datafactory"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataFactoryV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataFactoryV2CreateOrUpdate,
+		Read:   resourceArmDataFactoryV2Read,
+		Update: resourceArmDataFactoryV2CreateOrUpdate,
+		Delete: resourceArmDataFactoryV2Delete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$`),
+					`Invalid name for Data Factory, see https://docs.microsoft.com/en-us/azure/data-factory/naming-rules`,
+				),
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								"SystemAssigned",
+							}, true),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"github_configuration": {
```

Comment:
You need to set personal_access_token for Data Factory to install web hooks on the repo.

Reply:
As noted on the linked rest-api-specs issue, there's no option to pass in an access token, it may be that GitHub configuration using the API doesn't work. I don't have access to GitHub enterprise to be able to test.

Reply:
![image001](https://user-images.githubusercontent.com/1684739/44369501-74b18300-a48b-11e8-9121-ad3fea1c4033.jpg)

@hbuckle see the snapshot. From the UI portal of Data Factory, you will need to provide the token to guarantee it to access repo and install web hooks there.

Reply:
Interesting, when I did it through the UI I was prompted to allow the AzureDataFactory OAuth application access to my account and then it worked - I was hoping that as long as that application can access the GitHub account then adding the config through the API would work but it doesn't seem to be the case.

Reply:
@hbuckle After sync up with ADF team, you need to use below API to set up configuration:
https://docs.microsoft.com/en-us/rest/api/datafactory/factories/configurefactoryrepo

And they don't support any authentication through API, UI portal did some extra work.

Reply:
I am thinking of just removing the github_configuration as it doens't look like it will work using the API

Reply:
I've updated the code to use configurefactoryrepo, but we've also discovered a casing issue in the specs so github configuration won't work until that filters down to the SDK